### ISSUE TITLE
Election fails if someone doesn't get elected on first prefs (I think...)

### ISSUE THREAD

**Issue by OlympusMonds** [michaelsproul/aus_senate#20]
Hey bud,

I get this error:
```
RUST_BACKTRACE=1 target/release/election2016 data/candidate_ordering.csv data/ACT.csv ACT 2 8
Num groups: 10
Excluding disqualified candidates
Count #1
Count #2
thread 'main' panicked at 'election should terminate before running out of preferences to transfer', libcore/option.rs:917:5
```
when I mess with peoples preferences in certain ways. I think when no-one is elected on first prefs, this function:
https://github.com/michaelsproul/aus_senate/blob/124762b0749da417aa12ebbf58cba0e6ecad40e7/src/vote_map.rs#L212
returns an empty vec to here:
https://github.com/michaelsproul/aus_senate/blob/b824d6ad6e0db483954cb501c3ab0d60283db01f/src/voting.rs#L91
and so `preference_transfers` cannot `pop_front`.

Not entirely sure though!

**Comment by michaelsproul** [comment]
Interesting! I suspect the correct solution is to exclude the last candidate and use their exclusion as the first event to process. I'll dig into the legislation to see if it says any different

**Comment by OlympusMonds** [comment]
OK, on my very quick reading of [point 13](http://www8.austlii.edu.au/cgi-bin/viewdoc/au/legis/cth/consol_act/cea1918233/s273.html), I think it's OK.

> (13)  Where, after the counting of first preference votes or the transfer of surplus votes (if any) of elected candidates, no candidate has, or fewer than the number of candidates required to be elected have, received a number of votes equal to the quota:
(a)  the candidate who stands lowest in the poll must be excluded; or
(b)  if a bulk exclusion of candidates may be effected under subsection (13A), those candidates must be excluded;
and the ballot papers of the excluded candidate or candidates must be distributed in accordance with subsection (13AA).

**Comment by michaelsproul** [comment]
Closed by #21

**Comment by OlympusMonds** [comment]
hell yea
